internal/file/file.go

```diff
@@ -0,0 +1,111 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/brexhq/substation/internal/aws/s3manager"
+	"github.com/brexhq/substation/internal/errors"
+	"github.com/brexhq/substation/internal/http"
+)
+
+var (
+	httpClient   http.HTTP
+	s3managerAPI s3manager.DownloaderAPI
+)
+
+// errEmptyFile is returned when Get is called but finds an empty file.
+const errEmptyFile = errors.Error("empty file found")
+
+// errNoFile is returned when Get is called but no file is found.
+const errNoFile = errors.Error("no file found")
+
+/*
+Get retrieves a file from these locations (in order):
+
+- Local disk
+
+- HTTP or HTTPS URL
+
+- AWS S3
+
+If a file is found, then it is saved to disk and the path is returned. The caller is responsible for removing files when they are no longer needed.
+*/
+func Get(ctx context.Context, path string) (string, error) {
+	file, err := os.CreateTemp("", "tempfile")
+	if err != nil {
+		return "", fmt.Errorf("file %s: %v", path, err)
+	}
+
+	if _, err := os.Stat(path); err == nil {
```

Comment:
I think this would hurt readability and we may need to refactor it later if we make this change. There are cases where we use switch statements [that improve readability](https://github.com/brexhq/substation/blob/main/process/case.go#L88-L97) and others where [readability suffers](https://github.com/brexhq/substation/blob/main/process/base64.go#L86-L102), and due to the complexity of what is happening under each condition I think readability would suffer here. We've also used switch when we need to update a variable that exists outside the statement -- since these conditions only return, I don't think we gain much by using a switch.

The refactoring part is due to conditions like `if _, err := os.Stat(path); err == nil {}` -- I've no idea how many more of these we may add in the future, and I don't think this can be put into a switch.